rskj-core/src/test/java/co/rsk/peg/BridgeStorageProviderTest.java

```diff
@@ -3308,6 +3308,200 @@ public void saveNextPegoutHeight_after_RSKIP271() {
+            repository, PrecompiledContracts.BRIDGE_ADDR,
+            config.getNetworkConstants().getBridgeConstants(), activationsBeforeFork
+        );
+
+        Sha256Hash hash = Sha256Hash.ZERO_HASH;
+
+        DataWord storageKeyForPegoutCreationIndex = DataWord.fromLongString("pegoutCreationIndex-" + hash);
+        when(
+            repository.getStorageBytes(
+                PrecompiledContracts.BRIDGE_ADDR,
+                storageKeyForPegoutCreationIndex
+            )
+        ).thenReturn(BridgeSerializationUtils.serializeKeccak256(Keccak256.ZERO_HASH));
+
+        assertEquals(Optional.empty(), provider.getPegoutCreationEntry(
+            hash
+        ));
+
+        verify(repository, never()).getStorageBytes(PrecompiledContracts.BRIDGE_ADDR, storageKeyForPegoutCreationIndex);
+    }
+
+    @Test
+    public void getPegoutCreationEntry_after_RSKIP298() {
+        Repository repository = mock(Repository.class);
+        BridgeStorageProvider provider = new BridgeStorageProvider(
+            repository, PrecompiledContracts.BRIDGE_ADDR,
+            config.getNetworkConstants().getBridgeConstants(), activationsAllForks
+        );
+
+        Sha256Hash hash = Sha256Hash.ZERO_HASH;
+        Keccak256 keccak256 = Keccak256.ZERO_HASH;
+
+        DataWord storageKeyForPegoutCreationIndex = DataWord.fromLongString("pegoutCreationIndex-" + hash);
+        when(
+            repository.getStorageBytes(
+                PrecompiledContracts.BRIDGE_ADDR,
+                storageKeyForPegoutCreationIndex
+            )
+        ).thenReturn(BridgeSerializationUtils.serializeKeccak256(keccak256));
+
+        Optional<Keccak256> pegoutCreationEntry = provider.getPegoutCreationEntry(
+            hash
+        );
+        assertTrue(pegoutCreationEntry.isPresent());
+
+        assertEquals(keccak256, pegoutCreationEntry.get());
+
+        verify(repository, times(1)).getStorageBytes(
+            PrecompiledContracts.BRIDGE_ADDR,
+            storageKeyForPegoutCreationIndex
+        );
+    }
+
+    @Test
+    public void setPegoutCreationEntry_before_RSKIP298() {
+        Repository repository = mock(Repository.class);
+
+        BridgeStorageProvider provider = new BridgeStorageProvider(
+            repository, PrecompiledContracts.BRIDGE_ADDR,
+            config.getNetworkConstants().getBridgeConstants(), activationsBeforeFork
+        );
+
+        Sha256Hash hash = Sha256Hash.ZERO_HASH;
+        Keccak256 keccak256 = Keccak256.ZERO_HASH;
+
+        DataWord storageKeyForPegoutCreationIndex = DataWord.fromLongString("pegoutCreationIndex-" + hash);
+
+        provider.setPegoutCreationEntry(new PegoutCreationEntry(hash, keccak256));
+        provider.savePegoutCreationEntry();
+
+        verify(repository, never()).addStorageBytes(
+            PrecompiledContracts.BRIDGE_ADDR,
+            storageKeyForPegoutCreationIndex,
+            BridgeSerializationUtils.serializeKeccak256(keccak256)
```

Comment:
I don't like this change. Remember, what happened with the security issue on flyover pegin, the test was exactly doing this. This may hide a wrong behavior as happened on flyover.

Reply:
Makes sense to me, thanks!